### ISSUE TITLE
ESLint: Failed to load plugin 'react' declared in '.eslintrc.json'

### ISSUE THREAD

**Issue by orzhtml** [eslint/eslint#14732]
**Tell us about your environment**
macbook pro / macOS Catalina v10.15.7
visual studio code v1.32.3

* **ESLint Version:** 7.29.0
* **Node Version:**v14.17.0
* **npm Version:** 7.17.0
* **Operating System:** macOS Catalina v10.15.7

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
    "@typescript-eslint/eslint-plugin": "^4.28.0",
    "@typescript-eslint/parser": "^4.28.0",
    "eslint": "^7.29.0",
    "eslint-plugin-react": "^7.24.0",
    "eslint-plugin-react-hooks": "^4.2.0"

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
{
  "env": {
    "browser": true,
    "es2021": true,
    "node": true,
    "es6": true,
    "mocha": true,
    "jasmine": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "ecmaVersion": 12,
    "sourceType": "module"
  },
  "plugins": ["react", "@typescript-eslint", "react-hooks"],
  "rules": {
    "react/jsx-filename-extension": [
      1,
      { "extensions": [".js", "jsx", ".ts", ".tsx"] }
    ]
  }
}

```

```bash

```

**What did you expect to happen?**

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

ESLint: Failed to load plugin 'react' declared in '.eslintrc.json': createRequire is not a function Referenced from: /Volumes/HDD/website/react-app/web-app-ts/.eslintrc.json. Please see the 'ESLint' output channel for details.

**Steps to reproduce this issue:**

**Are you willing to submit a pull request to fix this bug?**

**Comment by orzhtml** [comment]
[Info  - 11:38:10 AM] ESLint server stopped.
[Info  - 11:38:11 AM] ESLint server running in node v10.2.0
[Info  - 11:38:11 AM] ESLint server is running.
[Info  - 11:38:13 AM] ESLint library loaded from: /Volumes/HDD/website/react-app/web-app-ts/node_modules/eslint/lib/api.js
[Error - 11:38:14 AM] ESLint stack trace:
[Error - 11:38:14 AM] TypeError: Failed to load plugin 'react' declared in '.eslintrc.json': createRequire is not a function
Referenced from: /Volumes/HDD/website/react-app/web-app-ts/.eslintrc.json
    at Object.resolve (/Volumes/HDD/website/react-app/web-app-ts/node_modules/@eslint/eslintrc/lib/shared/relative-module-resolver.js:28:20)
    at ConfigArrayFactory._loadPlugin (/Volumes/HDD/website/react-app/web-app-ts/node_modules/@eslint/eslintrc/lib/config-array-factory.js:1017:39)
    at ConfigArrayFactory._loadExtendedPluginConfig (/Volumes/HDD/website/react-app/web-app-ts/node_modules/@eslint/eslintrc/lib/config-array-factory.js:837:29)
    at ConfigArrayFactory._loadExtends (/Volumes/HDD/website/react-app/web-app-ts/node_modules/@eslint/eslintrc/lib/config-array-factory.js:779:29)
    at ConfigArrayFactory._normalizeObjectConfigDataBody (/Volumes/HDD/website/react-app/web-app-ts/node_modules/@eslint/eslintrc/lib/config-array-factory.js:720:25)
    at _normalizeObjectConfigDataBody.next (<anonymous>)
    at ConfigArrayFactory._normalizeObjectConfigData (/Volumes/HDD/website/react-app/web-app-ts/node_modules/@eslint/eslintrc/lib/config-array-factory.js:665:20)
    at _normalizeObjectConfigData.next (<anonymous>)
    at ConfigArrayFactory.loadInDirectory (/Volumes/HDD/website/react-app/web-app-ts/node_modules/@eslint/eslintrc/lib/config-array-factory.js:511:28)
    at CascadingConfigArrayFactory._loadConfigInAncestors (/Volumes/HDD/website/react-app/web-app-ts/node_modules/@eslint/eslintrc/lib/cascading-config-array-factory.js:379:46)

**Comment by aladdin-add** [comment]
eslint v7 supports node.js `^10.12.0 || >=12.0.0`, but seems you are using node.js v10, not"**Node Version:**v14.17.0" as you described in the issue.

so, can you confirm which node.js version you are using?

**Comment by orzhtml** [comment]
> eslint v7 supports node.js `^10.12.0 || >=12.0.0`, but seems you are using node.js v10, not"**Node Version:**v14.17.0" as you described in the issue.
> 
> so, can you confirm which node.js version you are using?

I don't have Node V10.2.0 locally

Visual Studio Code comes with NodeJS?

**Comment by aladdin-add** [comment]
sounds like that way. you can run command line(e.g. `./node_modules/.bin/eslint .`), if working fine, it would be something related to vscode.

**Comment by orzhtml** [comment]
> sounds like that way. you can run command line(e.g. `./node_modules/.bin/eslint .`), if working fine, it would be something related to vscode.

update Visual Studio Code v1.57.1  Ok！thank you~

![WX20210622-144135](https://user-images.githubusercontent.com/5866002/122876331-f75b9a80-d367-11eb-99ba-62319bb0f6e4.png)
